pkg/generic/thrift/read.go

```diff
@@ -336,6 +340,20 @@ func readStruct(ctx context.Context, in thrift.TProtocol, t *descriptor.TypeDesc
 	// void is nil struct
 	// default value with struct NOT SUPPORT pb.
 	if t.Struct != nil {
+		// set all fields even if it is empty, to be compatible with code-gen
+		if opt.setFieldsForEmptyStruct != 0 {
+			for _, field := range t.Struct.FieldsByID {
+				if opt.setFieldsForEmptyStruct == 1 {
+					// ignore optional fields if setFieldsForEmptyStruct == 1
+					if field.Optional {
+						continue
+					}
+				}
+				if err := fs(field, readEmptyValue(field.Type, opt)); err != nil {
```

Comment:
Though it may be trivial, if a number more than 2 is specified by users, are all the fields set?

Reply:
yes，I think UB is OK